src/dotnet-install.sh

```diff
@@ -713,44 +732,56 @@ download() {
     return 0
 }
 
+# Note, function writes to global variables $http_code and $download_error_msg
 downloadcurl() {
     eval $invocation
     local remote_path="$1"
     local out_path="${2:-}"
-
     # Append feed_credential as late as possible before calling curl to avoid logging feed_credential
-    remote_path="${remote_path}${feed_credential}"
-
+    local remote_path_with_credential="${remote_path}${feed_credential}"
     local curl_options="--retry 20 --retry-delay 2 --connect-timeout 15 -sSL -f --create-dirs "
     local failed=false
     if [ -z "$out_path" ]; then
-        curl $curl_options "$remote_path" || failed=true
+        curl $curl_options "$remote_path_with_credential" || failed=true
     else
-        curl $curl_options -o "$out_path" "$remote_path" || failed=true
+        curl $curl_options -o "$out_path" "$remote_path_with_credential" || failed=true
     fi
     if [ "$failed" = true ]; then
-        say_verbose "Curl download failed"
+        local response=$(get_http_header_curl $remote_path_with_credential)
+        http_code=$( echo "$response" | awk '/^HTTP/{print $2}' | tail -1 )
+        download_error_msg="Curl download $remote_path failed."
+        if  [[ $http_code != 2* ]]; then
+            download_error_msg+=" Returned HTTP status code: $http_code."
+        fi
+        say_verbose "$download_error_msg"
         return 1
     fi
     return 0
 }
 
+
+# Note, function writes to global variables $http_code and $download_error_msg
 downloadwget() {
     eval $invocation
     local remote_path="$1"
     local out_path="${2:-}"
-
     # Append feed_credential as late as possible before calling wget to avoid logging feed_credential
-    remote_path="${remote_path}${feed_credential}"
+    local remote_path_with_credential="${remote_path}${feed_credential}"
     local wget_options="--tries 20 --waitretry 2 --connect-timeout 15 "
     local failed=false
     if [ -z "$out_path" ]; then
-        wget -q $wget_options -O - "$remote_path" || failed=true
+        wget -q $wget_options -O - "$remote_path_with_credential" || failed=true
     else
-        wget $wget_options -O "$out_path" "$remote_path" || failed=true
+        wget $wget_options -O "$out_path" "$remote_path_with_credential" || failed=true
     fi
     if [ "$failed" = true ]; then
-        say_verbose "Wget download failed"
+        local response=$(get_http_header_wget $remote_path_with_credential)
```

Comment:
I could not get both the HTTP response code and the output of the download together with one request safely (as they could both go to standard output and I consider parsing it in order to separate them too risky), so I decided to ask for a header and get the HTTP response code from there in case of failure. Likely, in most of the cases we will have the same response. I am open to converse about that decision.

Reply:
I am a bit worried about this. Under heavy load, for instance, server could process one of the requests, but not the other. Is there no way to get the content and the response code separately?

Reply:
If it processes the real request, the second one will not be called. If the real one failed and the request for the header is successful this bit of information will not show. I have not found a way to separate these two outputs except making two separate requests. We still can get both of them in stdout and try to parse the result, but I do not like that idea more than the implemented one. I have not found a way to properly separate the content and the response code. If one suggests the way how to get them separately (when the content goes to stdout), I will be glad to consider that.